app/liquid_tags/unified_embed/tag.rb

```diff
@@ -21,21 +23,42 @@ class Tag < LiquidTagBase
     # @return [LiquidTagBase]
     def self.new(tag_name, link, parse_context)
       stripped_link = ActionController::Base.helpers.strip_tags(link).strip
-      klass = UnifiedEmbed::Registry.find_liquid_tag_for(link: stripped_link)
-      # If we can't find a registered "embed" tag, let's raise an exception.
-      # This exception will give the user an opportunity to adjust their approach.
-      #
-      # In a prior implementation, we chose to render an A-tag using the given URL.
-      # With that prior implementation, a user expecting a "rich embed" might not
-      # notice that they didn't have a rich embed and instead published a basic
-      # A-tag. In addition, said A-tag would goes nowhere; which may confuse
-      # users and/or Forem readers.
-      raise StandardError, I18n.t("liquid_tags.unified_embed.tag.invalid_url") unless klass
+
+      # Before matching against the embed registry, we check if the link
+      # is valid (e.g. no typos).
+      # If the link is invalid, we raise an error encouraging the user to
+      # check their link and try again.
+      validated_link = validate_link(stripped_link)
+      klass = UnifiedEmbed::Registry.find_liquid_tag_for(link: validated_link)
+
+      # If the link is valid but doesn't match the registry, we return
+      # an "unsupported URL" error. Eventually we shall render a fallback
+      # embed using OpenGraph/TwitterCard metadata (if available).
+      # If there are no OG metatags, then we render an A-tag. Since the link
+      # has been validated, at least this A-tag will not 404.
+      raise StandardError, I18n.t("liquid_tags.unified_embed.tag.unsupported_url") unless klass
 
       # Why the __send__?  Because a LiquidTagBase class "privatizes"
       # the `.new` method.  And we want to instantiate the specific
       # liquid tag for the given link.
-      klass.__send__(:new, tag_name, stripped_link, parse_context)
+      klass.__send__(:new, tag_name, validated_link, parse_context)
+    end
+
+    def self.validate_link(link)
+      uri = URI.parse(link)
+      http = Net::HTTP.new(uri.host, uri.port)
+      http.use_ssl = true if http.port == 443
+      path = uri.path.presence || "/"
+      response = http.request_head(path)
+
+      unless response.is_a?(Net::HTTPSuccess)
```

Comment:
we may want to consider checking for `Net::HTTPNotFound` in addition to this. Only because I'm not entirely sure how it will handle a redirect. Essentially, if a provided url gets a 301 will it follow that redirect THEN return with 200? If so, we don't need to worry about it. If not, we'll have to add in maybe a couple more explicit checks here.

Reply:
Perhaps we can test a redirect response to see?